client/components/data/query-share-post-actions/index.jsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import isFetchingPublicizeShareActionsScheduled from 'state/selectors/is-fetching-publicize-share-actions-scheduled';
+import isFetchingPublicizeShareActionsPublished from 'state/selectors/is-fetching-publicize-share-actions-published';
+import { fetchPostShareActionsScheduled, fetchPostShareActionsPublished } from 'state/sharing/publicize/publicize-actions/actions';
+
+class QuerySharePostActions extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		postId: PropTypes.number,
+		status: PropTypes.string,
+		// Connected props
+		isRequestingScheduled: PropTypes.bool.isRequired,
+		isRequestingPublished: PropTypes.bool.isRequired,
+		fetchPostShareActionsScheduled: PropTypes.func.isRequired,
+		fetchPostShareActionsPublished: PropTypes.func.isRequired,
+	};
+
+	componentDidMount() {
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
```

Comment:
To safeguard against upcoming react changes, this should be `componentDidUpdate`. See the [cheat sheet](https://gist.github.com/bvaughn/923dffb2cd9504ee440791fade8db5f9)